flashing/flash_firmware.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+
+from pathlib import Path
+
+from pyocd.core.helpers import ConnectHelper
+from pyocd.core.target import Target
+from pyocd.flash.file_programmer import FileProgrammer
+
+from mac_converter import MacConverter
+
+parser = argparse.ArgumentParser()
```

Comment:
Nit: I would simplify the CLI interface:
* -f for firmware flashing
* -s for softdevice flashing
* -m try to show the flashed MAC address (although please see my comment about the stability of this operation - if in doubt we can remove this feature)

I find that introducing -a and --only-mac creates an unnecessary cognitive load.

Reply:
Ok thanks for your input here. I wasn't quite sure about the "final" api. For me I really liked that I could only query the mac address via the same script but that was my personal use case. I am fine with removing it.

I propose a clearer approach by optional flags to provide the path to the softdevice and firmware file and optional flags to skip either.

The default case is then to make a full flash with everything but leaving you the option to skip the softdevice which doesn't change alot but still reflashing a new firmware build. 😃